go-controller/hybrid-overlay/pkg/controller/master.go

```diff
@@ -165,7 +165,25 @@ func (m *MasterController) releaseNodeSubnet(nodeName string, nodeSubnet *net.IP
 	return fmt.Errorf("failed to delete subnet %s for node %q: subnet not found in any CIDR range or already available", nodeSubnet, nodeName)
 }
 
-func (m *MasterController) handleOverlayPort(node *kapi.Node, annotator kube.Annotator) error {
+func (m *MasterController) deletePodsWithHybridOverlayIP(ip net.IP) error {
+	pods, err := m.kube.GetPods("")
+	if err != nil {
+		return fmt.Errorf("failed to get all pods: %v", err)
+	}
+
+	for _, pod := range pods.Items {
+		if podInfo, _ := util.UnmarshalPodAnnotation(pod.Annotations["ovn"]); err != nil {
+			if podInfo.IP.IP.Equal(ip) {
+				err := m.kube.DeletePod(pod.Namespace, pod.Name)
```

Comment:
This is pretty violent. At least in openshift-sdn when we kill pods, they'll still show up in `kubectl get pods` in an Error state. With this code, they're just gone. (Sure, *probably* you have a Deployment or ReplicaSet or something managing them, but...)

And do we really need to support this upgrade case anyway?

Reply:
@danwinship yep, pretty violent. But we don't really have any mechanism to enable this at install time, so you'll always have a cluster with it turned off and potentially have pods with the .3 address.

Alternatively, we can always exclude the .3 address on the node switch, but then the moment people upgrade to this version that does exclude it, those pods get re-IPAM-ed by OVN and are then out-of-sync and must be killed :(

So yeah, we kinda need to do this for the upgrade case.

Reply:
Decided to just always exclude the 3rd IP if all cluster subnets are large enough, which avoids the upgrade problems and necessity to kill pods using the 3rd IP.